File: Mobile/PlatziFlixiOS/PlatziFlixiOS/Data/Entities/RatingDTO.swift
import Foundation

struct RatingRequestDTO: Encodable {
    let userId: Int
    let rating: Int

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case rating
    }
}

struct RatingResponseDTO: Codable {
    let id: Int
    let courseId: Int
    let userId: Int
    let rating: Int
    let createdAt: String
    let updatedAt: String

    enum CodingKeys: String, CodingKey {
        case id
        case courseId = "course_id"
        case userId = "user_id"
        case rating
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

struct RatingStatsDTO: Codable {
    let averageRating: Double
    let totalRatings: Int
    let ratingDistribution: [String: Int]

    enum CodingKeys: String, CodingKey {
        case averageRating = "average_rating"
        case totalRatings = "total_ratings"
        case ratingDistribution = "rating_distribution"
    }
}

File: Mobile/PlatziFlixAndroid/app/src/main/java/com/espaciotiago/platziflixandroid/data/entities/RatingDTO.kt
package com.espaciotiago.platziflixandroid.data.entities

import com.google.gson.annotations.SerializedName

data class RatingRequestDTO(
    @SerializedName("user_id") val userId: Int,
    @SerializedName("rating") val rating: Int
)

data class RatingResponseDTO(
    @SerializedName("id") val id: Int,
    @SerializedName("course_id") val courseId: Int,
    @SerializedName("user_id") val userId: Int,
    @SerializedName("rating") val rating: Int,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String
)

data class RatingStatsDTO(
    @SerializedName("average_rating") val averageRating: Double,
    @SerializedName("total_ratings") val totalRatings: Int,
    @SerializedName("rating_distribution") val ratingDistribution: Map<String, Int>
)
